wait.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+	"time"
+)
+
+// Wait for hosts to become ready
+func Wait(hosts []string, timeout int) {
+	begin := time.Now()
+	var wg sync.WaitGroup
+	cancel := make(chan struct{})
+
+	for _, host := range hosts {
+		wg.Add(1)
+		go func(host string) {
+			WaitForHTTPRequest(host, cancel)
+			wg.Done()
+		}(fmt.Sprintf("%s:8080", host))
+	}
+
+	timer := time.AfterFunc(time.Duration(timeout)*time.Second, func() {
+		close(cancel)
+	})
+
+	wg.Wait()
+
+	if !timer.Stop() {
+		log.Printf("Error: One or more services timed out after %d second(s)", timeout)
+		os.Exit(1)
+	}
+	fmt.Printf("\nAll services are up after %v!\n", time.Now().Sub(begin))
+}
+
+// WaitForHTTPRequest polls host until it can make a request
+func WaitForHTTPRequest(host string, cancel <-chan struct{}) {
+	url := url.URL{
+		Scheme: "http",
+		Host:   host,
+		Path:   "/",
+	}
+
+	err := errors.New("init")
+	for err != nil {
+		req, reqErr := http.NewRequest("GET", url.String(), nil)
+		if reqErr != nil {
+			log.Printf("Warning: Failed to create request for URL '%s' -  skipping service '%s'",
+				url.String(), host)
+			return
+		}
+
+		tr := &http.Transport{}
```

Comment:
`DefaultTransport` is just a transport in the `http` package,
https://golang.org/pkg/net/http/#DefaultTransport

But @abhinav has a good point, you should just use the `ctxhttp` package to do this for you. It requires a `*http.Client` where you can just pass `http.DefaultClient`